tests/director/tests/formation_constraint_api_test.go

```diff
@@ -287,9 +287,27 @@ func TestListFormationConstraintsForFormationTemplate(t *testing.T) {
 	defer fixtures.CleanupFormationConstraint(t, ctx, certSecuredGraphQLClient, constraint.ID)
 	require.NotEmpty(t, constraint.ID)
 
+	// Assert no constraints attached
+	t.Logf("Get formation template with name %q and id %q, and assert there are no constraints attached to it", formationTemplate.Name, formationTemplate.ID)
+	formationTemplateOutput := fixtures.QueryFormationTemplateWithConstraints(t, ctx, certSecuredGraphQLClient, formationTemplate.ID)
+	assert.Empty(t, formationTemplateOutput.FormationConstraints)
+
+	t.Logf("Get formation template with name %q and id %q, and assert there are no constraints attached to it", secondFormationTemplate.Name, secondFormationTemplate.ID)
+	secondFormationTemplateOutput := fixtures.QueryFormationTemplateWithConstraints(t, ctx, certSecuredGraphQLClient, secondFormationTemplate.ID)
+	assert.Empty(t, secondFormationTemplateOutput.FormationConstraints)
+
 	t.Logf("Attaching constraint to formation template")
 	fixtures.AttachConstraintToFormationTemplate(t, ctx, certSecuredGraphQLClient, constraint.ID, formationTemplate.ID)
 
+	// Assert the constraint is attached only to the first formation template
+	t.Logf("Get formation template with name %q and id %q, and assert there are is one constraint attached to it", formationTemplate.Name, formationTemplate.ID)
+	formationTemplateOutput = fixtures.QueryFormationTemplateWithConstraints(t, ctx, certSecuredGraphQLClient, formationTemplate.ID)
+	assert.Equal(t, formationTemplateOutput.FormationConstraints, []graphql.FormationConstraint{*constraint})
+
+	t.Logf("Get formation template with name %q and id %q, and assert there are no constraints attached to it", secondFormationTemplate.Name, secondFormationTemplate.ID)
+	secondFormationTemplateOutput = fixtures.QueryFormationTemplateWithConstraints(t, ctx, certSecuredGraphQLClient, secondFormationTemplate.ID)
+	assert.Empty(t, secondFormationTemplateOutput.FormationConstraints)
```

Comment:
Do we need this? As we already check that a few lines above and the "attach" operation before this assert is related to the "first" formation template, we do not expect the "second" one to be changed?

Reply:
It can be removed but with this, we are checking that the sub-resolver is working correctly(that it returns the correct constraints for the right formation templates).